src/test/java/rx/observables/BlockingSingleTest.java

```diff
@@ -0,0 +1,83 @@
+package rx.observables;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import rx.Single;
+import rx.Subscriber;
+import rx.exceptions.TestException;
+
+import java.util.concurrent.Future;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test suite for {@link BlockingSingle}.
+ */
+public class BlockingSingleTest {
+
+    @Mock
+    Subscriber<Integer> w;
+
+    @Before
+    public void before() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testSingleGet() {
+        Single<String> single = Single.just("one");
+        BlockingSingle<? extends String> blockingSingle = BlockingSingle.from(single);
+        assertEquals("one", blockingSingle.value());
+    }
+
+    @Test
+    public void testSingleError() {
+        Single<String> single = Single.error(new TestException());
+        BlockingSingle<? extends String> blockingSingle = BlockingSingle.from(single);
+
+        Exception caughtException = null;
+        try {
+            blockingSingle.value();
+        } catch (Exception e) {
+            caughtException = e;
+        }
+        assertNotNull(caughtException);
+        assertTrue(caughtException instanceof TestException);
+    }
+
+    @Test
+    public void testSingleErrorChecked() {
+        Single<String> single = Single.error(new TestCheckedException());
+        BlockingSingle<? extends String> blockingSingle = BlockingSingle.from(single);
+
+        Exception caughtException = null;
+        try {
+            blockingSingle.value();
```

Comment:
Same about `fail()` and moving everything to the `catch` block, though it's fully up to you :)